src/status_im/contexts/settings/wallet/events.cljs

```diff
@@ -170,3 +200,36 @@
                :text  (:error error-data)}]]]})))
 
 (rf/reg-event-fx :wallet/import-keypair-by-seed-phrase-failed import-keypair-by-seed-phrase-failed)
+
+(defn import-keypair-by-private-key
+  [_ [{:keys [keypair-key-uid private-key password on-success on-error]}]]
+  {:fx [[:import-keypair-by-private-key
+         {:private-key private-key
+          :password    password
+          :on-success  (fn []
+                         (rf/dispatch [:wallet/make-keypairs-accounts-fully-operable
+                                       #{keypair-key-uid}])
+                         (cond
+                           (vector? on-success) (rf/dispatch (conj on-success))
+                           (fn? on-success)     (on-success)))
+          :on-error    (fn [error]
+                         (rf/dispatch [:wallet/import-keypair-by-private-key-failed error])
+                         (cond
+                           (vector? on-error) (rf/dispatch (conj on-error error))
+                           (fn? on-error)     (on-error error)))}]]})
+
+(rf/reg-event-fx :wallet/import-keypair-by-private-key import-keypair-by-private-key)
```

Comment:
Suggestion: Renaming to `:wallet/import-missing-keypair-type-private-key` or something to prevent overlapping with the actual import of keypairs using private key in add key pair flows

Reply:
yeah good point 👍